### PR TITLE
Fix style merging for overlay arrow

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Found in testing session

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by LFDanLu — APPROVED** [review]
LGTM, confirmed that the arrow is now fixed in the docs and other RAC examples

File: packages/react-aria-components/src/OverlayArrow.tsx [hunk 1]
```diff
@@ -48,6 +48,11 @@ function OverlayArrow(props: OverlayArrowProps, ref: ForwardedRef<HTMLDivElement
       placement
     }
   });
+  // remove undefined values from renderProps.style object so that it can be
+  // spread merged with the other style object
+  if (renderProps.style) {
+    Object.keys(renderProps.style).forEach(key => renderProps.style![key] === undefined && delete renderProps.style![key]);
+  }
```
Comment (devongovett) [inline packages/react-aria-components/src/OverlayArrow.tsx:55]: How did the undefineds get there? I didn't see any style prop in the examples. Will we need this in other places?

Comment (snowystinger) [reply packages/react-aria-components/src/OverlayArrow.tsx:55]: It just comes from here: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/overlays/src/calculatePosition.ts#L384
and then it's reinforced here
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/overlays/src/useOverlayPosition.ts#L224
so even if you removed it from the first one, it'd reappear there.

Now that we don't use mergeProps for everything in RAC, we will need to be careful about this.
Do you have any thoughts or preferences? I am happy to fix it upstream instead. I was just going for smallest change here.

Comment (devongovett) [reply packages/react-aria-components/src/OverlayArrow.tsx:55]: mergeProps doesn't merge styles either, so would have to be a new util I guess. fine for now

**Review by devongovett — APPROVED** [review]

**Pull request merged**